appservice/src/intTest/groovy/nu/yona/server/ActivityTest.groovy

```diff
@@ -460,6 +460,86 @@ class ActivityTest extends AbstractAppServiceIntegrationTest
 		appService.deleteUser(bob)
 	}
 
+	def 'Richard retrieves buddy activity info before Bob accepted Richard\'s buddy request'()
+	{
+		given:
+		User richard = addRichard()
+		User bob = addBob()
+		appService.sendBuddyConnectRequest(richard, bob)
+		richard = appService.reloadUser(richard)
+
+		when:
+		def responseDayOverviewsWithBuddies = appService.getDayActivityOverviewsWithBuddies(richard, ["size": 14])
+
+		then:
+		richard.buddies[0].dailyActivityReportsUrl == null
+		richard.buddies[0].weeklyActivityReportsUrl == null
+		responseDayOverviewsWithBuddies.status == 200
+		responseDayOverviewsWithBuddies.responseData._embedded."yona:dayActivityOverviews"[0].dayActivities.find{ it._links."yona:activityCategory"?.href == GAMBLING_ACT_CAT_URL}.dayActivitiesForUsers.size() == 1
+
+		cleanup:
+		appService.deleteUser(richard)
+		appService.deleteUser(bob)
+	}
+
+	def 'Richard retrieves buddy activity info before he processed Bob\'s acceptance'()
```

Comment:
Maybe it is also good to add a test for the scenario that Bob removed the buddy entry, or that Richard removed the buddy entry before the process?